src/tox/session.py

```diff
@@ -478,6 +477,37 @@ def _makesdist(self):
                 )
                 raise SystemExit(1)
 
+    def _make_package(self, action, setup):
+        self.make_emptydir(self.config.distdir)
+        if self.config.build == "sdist":
+            action.popen(
+                [
+                    sys.executable,
+                    setup,
+                    "sdist",
+                    "--formats=zip",
+                    "--dist-dir",
+                    self.config.distdir,
+                ],
+                cwd=self.config.setupdir,
+            )
+        elif self.config.build == "wheel":
+            action.popen(
+                [
+                    sys.executable,
+                    "-m",
+                    "pip",
+                    "wheel",
+                    "-w",
+                    self.config.distdir,  # target folder
+                    "--no-deps",  # don't build wheels for package dependencies
+                    self.config.setupdir,  # what to build - ourselves
+                ],
+                cwd=self.config.setupdir,
+            )
+        else:
+            raise RuntimeError("invalid build type {}".format(self.config.build))
```

Comment:
Again the creation of a build venv is the responsibility of pip, and in case of the wheel it already does that. For doing the same for ``sdist`` the solution will be to move to ``pip build sdist`` and we'll support implementing PEP-517 inside pip. tox will not implement what you're describing as it's outside the scope of the tool.

Reply:
I'm sorry for breaking into your conversation but I have one argument to invoke ``pip wheel`` with virtualenv's python - packages with C extensions. They depend on Python version with which they are built and afaiu pip will build wheels only for interpeter it's being run by, so such wheels built by tox's interpreter will not suit environment with another one.

Reply:
That’s a valid point. Maybe tox should run pip wheel multiple times for each python version, when it isn’t a universal pure python wheel.

Reply:
Makes sense, thanks for the clarification